lib/Echidna/Processor.hs

```diff
@@ -0,0 +1,174 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Echidna.Processor where
+
+import Control.Arrow              (second)
+import Control.Lens
+import Control.Monad.IO.Class     (MonadIO(..))
+import Control.Exception          (Exception)
+import Control.Monad.Catch        (MonadThrow(..))
+import Data.Aeson                 (decode, Value(..))
+import Data.Text                  (Text, pack, unpack)
+import System.Directory           (findExecutable)
+import System.Process             (StdStream(..), readCreateProcessWithExitCode, proc, std_err)
+import System.Exit                (ExitCode(..))
+import GHC.Word                   (Word32)
+
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import qualified Data.HashMap.Strict as M
+
+import Echidna.ABI (hashSig)
+
+-- | Things that can go wrong trying to run a processor. Read the 'Show'
+-- instance for more detailed explanations.
+data ProcException = ProcessorFailure String String
```

Comment:
This will be used very soon when Slither is fully integrated with Echidna. Right now, since it is optional, we ignore any failure. My idea was to have a release where Slither is optional, and the next one will require it.